script/upgrades/MU04/MU04Checks.verify.sol

```diff
@@ -0,0 +1,323 @@
+      Config.Pool memory poolConfig = config.pools[i];
+
+      bytes32 asset0LimitId = exchangeId ^ bytes32(uint256(uint160(config.pools[i].asset0)));
+      TradingLimits.Config memory asset0ActualLimit = _broker.tradingLimitsConfig(asset0LimitId);
+
+      bytes32 asset1LimitId = exchangeId ^ bytes32(uint256(uint160(config.pools[i].asset1)));
+      TradingLimits.Config memory asset1ActualLimit = _broker.tradingLimitsConfig(asset1LimitId);
+
+      checkTradingLimt(poolConfig.asset0limits, asset0ActualLimit);
+      checkTradingLimt(poolConfig.asset1limits, asset1ActualLimit);
+    }
+
+    console.log("🟢 Trading limits correctly updated for all exchanges 🔒");
+  }
+
+  function checkTradingLimt(
+    Config.TradingLimit memory expectedTradingLimit,
+    TradingLimits.Config memory actualTradingLimit
+  ) internal view {
+    if (expectedTradingLimit.limit0 != actualTradingLimit.limit0) {
+      console.log("limit0 was not set as expected ❌");
+      revert("Not all trading limits were configured correctly.");
+    }
+    if (expectedTradingLimit.limit1 != actualTradingLimit.limit1) {
+      console.log("limit1 was not set as expected ❌");
+      revert("Not all trading limits were configured correctly.");
+    }
+    if (expectedTradingLimit.limitGlobal != actualTradingLimit.limitGlobal) {
+      console.log("limitGlobal was not set as expected ❌");
+      revert("Not all trading limits were configured correctly.");
+    }
+    if (expectedTradingLimit.timeStep0 != actualTradingLimit.timestep0) {
+      console.log("timestep0 was not set as expected ❌");
+      revert("Not all trading limits were configured correctly.");
+    }
+    if (expectedTradingLimit.timeStep1 != actualTradingLimit.timestep1) {
+      console.log("timestep1 was not set as expected ❌");
+      revert("Not all trading limits were configured correctly.");
+    }
+
+    uint8 tradingLimitFlags = Config.tradingLimitConfigToFlag(expectedTradingLimit);
+    if (tradingLimitFlags != actualTradingLimit.flags) {
+      console.log("flags were not set as expected ❌");
+      revert("Not all trading limits were configured correctly.");
+    }
+  }
+
+  function bytes32ToStr(bytes32 _bytes32) public view returns (string memory) {
+    uint256 length = 0;
+    while (bytes1(_bytes32[length]) != 0) {
+      length++;
+    }
+
+    bytes memory bytesArray = new bytes(length);
+    for (uint256 i; i < length; i++) {
+      bytesArray[i] = _bytes32[i];
+    }
+    return string(bytesArray);
+  }
+
+  function verifyCollateralSpendingRatio(address collateralAsset, uint256 expectedRatio) internal {
+    console.log("\n== Verifying Collateral Spending Ratio for %s ==", collateralAsset);
+    // @notice verifiying spending ratios by trying moving the allowed and more than the allowed amount
+    // the variable holding the ratios is private
+
+    address[] memory otherReserveAddresses = Reserve(reserveProxy).getOtherReserveAddresses();
+    if (otherReserveAddresses.length > 0) {
```

Comment:
is there a network for which this is not true? If not, let's change this to a require instead?

Reply:
I haven't checked but let me find out and if all of them have another reserve I will remove it 👍